### PR TITLE
AWS: TargetResponseTime metrics added to Application elb

### LINKED ISSUE #9201: Add TargetResponseTime metric in ELB agent configuration,
While working with [AWS ELB integration](https://docs.elastic.co/en/integrations/aws/elb) customers would wish to know the Latency for the application load balancers. 

Observation was that we are currently unable to see any data points for the metric on the default dashboards that are provided with the integration and I understand that we are making use of: `aws.elb.metrics.Latency.avg` field.  

From AWS's end we have `[TargetResponseTime](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-cloudwatch-metrics.html)` which we would like to have in the integration.

### PR DESCRIPTION
**Pull request by anil-elastic**
## Proposed commit message
Added TargetResponseTime metrics in AWS Application ELB.
 

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## How to test this PR locally
I've tested with elastic-package, and run the tests and  TargetResponseTime comes into elb metrics.

```
"aws": {
      "applicationelb": {
        "metrics": {
          "HealthyHostCount": {
            "max": 1
          },
          "RequestCount": {
            "sum": 4
          },
          "TargetResponseTime": {
            "avg": 0.0014383333333333333
          },
          "UnHealthyHostCount": {
            "max": 0
          }
        }
      },
```

- Closes #9201

### CODE REVIEW

**Comment by anil-elastic** [comment]
Signed the Contributor Agreement.

**Comment by agithomas** [comment]
I understand this is a simple enhancement. 

Is there a plan to display the newly added field under the existing dashboard? 


This change must be considered alongside other issues mentioned [here](https://github.com/elastic/integrations/issues/10151), especially the challenges of dashboarding for ELB metrics and logs.

File: packages/aws/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "2.15.5"
+  changes:
+    - description: Add TargetResponseTime metric and implements avg aggregation for existing elb application.
+      type: bugfix
```
Comment (agithomas) [inline packages/aws/changelog.yml:5]: This is an enhancement. Kindly assign the appropriate type.

Please correct the PR link

File: packages/aws/data_stream/elb_metrics/agent/stream/stream.yml.hbs [hunk 2]
```diff
@@ -100,6 +100,7 @@ metrics:
   statistic: ["Average"]
   name:
   - ConsumedLCUs
+  - TargetResponseTime
```
Comment (agithomas) [inline packages/aws/data_stream/elb_metrics/agent/stream/stream.yml.hbs:103]: Why `Average` statistic is considered? Why this is preferred over `Maximum` or `Minimum` ?

File: packages/aws/data_stream/elb_metrics/fields/fields.yml [hunk 3]
```diff
@@ -161,6 +161,10 @@
               type: double
               metric_type: gauge
               description: The number of load balancer capacity units (LCU) used by your load balancer.
+            - name: TargetResponseTime.avg
```
Comment (agithomas) [inline packages/aws/data_stream/elb_metrics/fields/fields.yml:164]: Please assign the correct type and unit suitable for Time based field.

File: packages/aws/manifest.yml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 format_version: 3.0.0
 name: aws
 title: AWS
-version: 2.15.4
+version: 2.15.5
```
Comment (agithomas) [inline packages/aws/manifest.yml:4]: As this is an enhancement please increment the minor version number.

**Review by agithomas — COMMENTED** [review]
Reviews shared.

**Comment by agithomas** [comment]
Please regenerate the README, checkin the README changes to avoid buildkite failure.

File: packages/aws/data_stream/elb_metrics/agent/stream/stream.yml.hbs (hunk 2 above)
Comment (gpop63) [reply packages/aws/data_stream/elb_metrics/agent/stream/stream.yml.hbs:103]: @agithomas The `Average` statistic is recommended by AWS for this metric.

> Statistics: The most useful statistics are Average and pNN.NN (percentiles).

https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-cloudwatch-metrics.html

File: packages/aws/data_stream/elb_metrics/fields/fields.yml (hunk 3 above)
Comment (anil-elastic) [reply packages/aws/data_stream/elb_metrics/fields/fields.yml:164]: @agithomas Are you saying that "double" is incorrect for this metric?
metric type is not date or time.

From the official doc:
**The time elapsed, in seconds, after the request leaves the load balancer until the target starts to send the response headers.**

File: packages/aws/manifest.yml (hunk 4 above)
Comment (anil-elastic) [reply packages/aws/manifest.yml]: updated with 2.16.0

File: packages/aws/changelog.yml (hunk 1 above)
Comment (anil-elastic) [reply packages/aws/changelog.yml]: both issues corrected.

File: packages/aws/manifest.yml (hunk 4 above)
Comment (anil-elastic) [reply packages/aws/manifest.yml]: updated with 2.16.0

File: packages/aws/data_stream/elb_metrics/fields/fields.yml (hunk 3 above)
Comment (agithomas) [reply packages/aws/data_stream/elb_metrics/fields/fields.yml:164]: Can we use `type: long` instead of `type: double`?

**Review by agithomas — COMMENTED** [review]
Comments added

File: packages/aws/data_stream/elb_metrics/fields/fields.yml (hunk 3 above)
Comment (anil-elastic) [reply packages/aws/data_stream/elb_metrics/fields/fields.yml:164]: updated with long.

File: packages/aws/changelog.yml [hunk 5]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "2.16.0"
+  changes:
+    - description: Add TargetResponseTime metric to ELB Application metrics.
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/12345
```
Comment (gpop63) [inline packages/aws/changelog.yml:6]: Add the correct PR id.

```suggestion
      link: https://github.com/elastic/integrations/pull/10181
```

**Comment by ishleenk17** [comment]
@lucabelluccini : I think this was the customer enhancement request you were discussing with me. 
The PR is already in works for adding the TargetResponseTime metric.

**Review by gpop63 — APPROVED** [review]

**Comment by anil-elastic** [comment]
Reply to @Agi comment: https://github.com/elastic/integrations/pull/10181#issuecomment-2177623131

I had discussion with @Agi and according to @agithomas  Customers are often getting confused with empty fields in their dashboard.There is ongoing issue to split elb data stream into 3 different data streams: application_elb, network_elb, and classic_elb as mentioned in this ticket https://github.com/elastic/integrations/issues/4800. 

I have also created backlog ticket for create a dashboard panel for ALB.
https://github.com/elastic/integrations/issues/10326

**Comment by agithomas** [comment]
> I understand this is a simple enhancement.
> 
> Is there a plan to display the newly added field under the existing dashboard?
> 
> This change must be considered alongside other issues mentioned [here](https://github.com/elastic/integrations/issues/10151), especially the challenges of dashboarding for ELB metrics and logs.





Based on my discussion with @anil-elastic , i understand the field is relevant when an ALB is configured.

Adding a dashboard panel at this stage will add more problems to the customers as mentioned [here](https://github.com/elastic/integrations/issues/10149). 

Hence the https://github.com/elastic/integrations/issues/10326 is created as a backlog issue, which can be taken up along with the enhancement request of [separate OOTB dashboards for various LBs](https://github.com/elastic/integrations/issues/5953)

**Review by agithomas — APPROVED** [review]
LGTM!

**Pull request merged**